clustermesh/clustermesh.go

```diff
@@ -2082,6 +2073,108 @@ func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
 	return nil
 }
 
+// Helm-based clustermesh connect/disconnect is only supported on Cilium
+// v1.14+ due to a lack of support in earlier versions for
+// autoconfigured certificates (tls.{crt,key}) for cluster
+// members when running in certgen (cronJob) PKI mode
+func (k *K8sClusterMesh) needsClassicMode(r *release.Release) (bool, error) {
+	if r == nil {
+		return false, fmt.Errorf("needs a valid helm release. got nil")
+	}
+
+	version := r.Chart.AppVersion()
+	semv, err := utils.ParseCiliumVersion(version)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse Cilium version: %w", err)
+	}
+	k.Log("✅ Detected Helm release with Cilium version %s", semv)
+
+	const ciliumHelmMinRev = "1.14.0"
+	cv := semver.MustParse(ciliumHelmMinRev)
+	if semv.Major == cv.Major && semv.Minor < cv.Minor {
+		k.Log("⚠️ Cilium Version is less than %s. Continuing in classic mode.", ciliumHelmMinRev)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (k *K8sClusterMesh) DisconnectWithHelm(ctx context.Context) error {
+	localRelease, err := getRelease(k.client.(*k8s.Client), k.params.Namespace)
+	if err != nil {
+		k.Log("❌ Unable to find Helm release for the target cluster")
+		return err
+	}
+	ok, err := k.needsClassicMode(localRelease)
+	if err != nil {
+		return err
+	}
+	if ok {
+		return k.Disconnect(ctx)
+	}
+
+	localClient, remoteClient, err := k.getClientsForConnect()
+	if err != nil {
+		return err
+	}
+	aiLocal, aiRemote, err := k.getAccessInfoForConnect(ctx, localClient, remoteClient)
+	if err != nil {
+		return err
+	}
+	if err = k.validateInfoForConnect(aiLocal, aiRemote); err != nil {
+		return err
+	}
+
+	// Get existing helm values for the local cluster
+	localHelmValues := localRelease.Config
+	// Expand those values to include the clustermesh configuration
```

Comment:
Another nitpick: what I was pointing out with this comment (and the similar one below), was that in this case we already expect to have a clustermesh config, so we are not "expanding" it but rather: `// Modify the clustermesh config to remove the intended cluster`

Reply:
Ok - I'll update. Tx for the review Andrew!